src/Monolog/Handler/ProcessableHandlerTrait.php

```diff
@@ -13,19 +13,18 @@
 
 use Monolog\ResettableInterface;
 use Monolog\Processor\ProcessorInterface;
+use Monolog\LogRecord;
 
 /**
  * Helper trait for implementing ProcessableInterface
  *
  * @author Jordi Boggiano <j.boggiano@seld.be>
- *
- * @phpstan-import-type Record from \Monolog\Logger
  */
 trait ProcessableHandlerTrait
 {
     /**
      * @var callable[]
-     * @phpstan-var array<ProcessorInterface|callable(Record): Record>
+     * @phpstan-var array<array<(callable(LogRecord): LogRecord)|ProcessorInterface>>
```

Comment:
While bumping to PHP 8.1, it would be great to stop using callable and always use \Closure instead. The new $callable(...) syntax makes it a no brainer. Closures are more performant, more secure (a closure cannot be unserialized, saving ourselves from related issues), and easier to check by the engine (think typed properties).

Reply:
Ah interesting, wasn't aware, thanks :)

It still makes it a bit harder to pass in an object with __invoke I guess? But let's give it a shot.

Reply:
> It still makes it a bit harder to pass in an object with __invoke I guess? But let's give it a shot.

No, you can turn an invokable object into a closure like any other callable:

```php
$myClosure = $myInvokableObject(...);
```